File: Spotify/SpotifySDK/ios-sdk-1.2.2/DemoProjects/SPTLoginSampleApp/SPTLoginSampleApp/ConnectButton.m
#import "ConnectButton.h"


@implementation ConnectButton

+ (instancetype)buttonWithType:(UIButtonType)buttonType
{
    ConnectButton *button = [super buttonWithType:buttonType];
    button.backgroundColor = [self customBackgroundColor];
    button.contentEdgeInsets = UIEdgeInsetsMake(11.75, 32.0, 11.75, 32.0);
    button.layer.cornerRadius = 20.0;
    button.translatesAutoresizingMaskIntoConstraints = NO;

    return button;
}

+ (UIColor *)customBackgroundColor
{
    static UIColor *color;

    if (!color)
        color = [UIColor colorWithRed:(29.0 / 255.0) green:(185.0 / 255.0) blue:(84.0 / 255.0) alpha:1.0];

    return color;
}

+ (UIColor *)customHighlightedBackgroundColor
{
    static UIColor *color;

    if (!color)
        color = [UIColor colorWithRed:(24.0 / 255.0) green:(172.0 / 255.0) blue:(77.0 / 255.0) alpha:1.0];

    return color;
}

- (void)setHighlighted:(BOOL)highlighted
{
    self.backgroundColor = highlighted ? [self.class customHighlightedBackgroundColor] : [self.class customBackgroundColor];
}

@end

File: Spotify/SpotifySDK/DemoApps/ios-sdk-1.2.2/DemoProjects/SPTLoginSampleAppSwift/SPTLoginSampleAppSwift/ConnectButton.swift
import UIKit

class ConnectButton: UIButton {

    fileprivate let buttonBackgroundColor =
        UIColor(red:(29.0 / 255.0), green:(185.0 / 255.0), blue:(84.0 / 255.0), alpha:1.0)
    fileprivate let titleAttributes: [NSAttributedString.Key: Any] = [
        .font: UIFont.systemFont(ofSize: UIFont.systemFontSize, weight: .heavy),
        .foregroundColor: UIColor.white,
        .kern: 2.0
    ]

    init(title: String) {
        super.init(frame: CGRect.zero)
        backgroundColor = buttonBackgroundColor
        contentEdgeInsets = UIEdgeInsets(top: 11.75, left: 32.0, bottom: 11.75, right: 32.0)
        layer.cornerRadius = 20.0
        translatesAutoresizingMaskIntoConstraints = false
        let title = NSAttributedString(string: title, attributes: titleAttributes)
        setAttributedTitle(title, for: .normal)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
